fecgov/fec-cms
# Feature/2562/embed bar charts election map homepage

## Summary
Embed bar charts and election map in homepage


- Resolves #2562
- Replaces WIP PR: #2715

Please not that this iteration of the home `/templates/home/home_page.html` extends  a new template called `home_base.html` instead of the  `base.html` which is extended by most of the other pages in home. This is done in order to Set the `BASE_PATH` variables  to `/data` ( it is set to  `/ ` in `base.html` )

## Impacted areas of the application
List general components of the application that this PR will affect:

**Modifies files**:    
        modified:   /settings/base.py
	modified:   fec/static/js/modules/top-entities.js
	modified:   fec/static/scss/components/_breakdowns.scss
	modified:   fec/static/scss/components/_icon-headings.scss
	modified:   fec/static/scss/components/_search-controls.scss
	modified:   fec/static/scss/components/_table-styles.scss
	modified:   fec/static/scss/layout/_slabs.scss
	modified:   home/templates/home/home_page.html

**New Files:**
	fec/templates/home_base.html
	home/templates/partials/elections-lookup.html
	home/templates/partials/raising-spending.html
	home/templatetags/elections_lookup.py
	home/templatetags/top_entities.py

## Screenshots

<kbd>

![embeds_scrnsht](https://user-images.githubusercontent.com/5572856/54502792-d1cac580-4902-11e9-8a5e-ecb863e2306d.jpg)


</kbd>


## Related PRs
List related PRs against other branches:

## How to test
Include any information that may be helpful to the reviewer(s).
  - Run npm `run-build`
 - Set feature flags in your shell session or in your bash profile to map and barcharts:
 `FEC_FEATURE_HOME_BARCHARTS=1` , `FEC_FEATURE_HOME_MAP=1`
- Visit http://127.0.0.1:8000/ and test the embedded map and barchart apps
- This branch is based on the new develop branch with node upgrade, so if you have not pulled the new develop (and updated your environment), you may need to follow some of the how-to-test  steps here  to get this to work:  https://github.com/fecgov/fec-cms/pull/2733

**For future reference, should we need to make data apps available to home templates:**
Because the data app uses jinja ( .jinja) templates and the home app uses django (.html) templates, making the bar-charts and election-map available to the homepage involves recreating the logic in views/macros in data as templatetags/inclusion-tags in home. There are other ways of doing this, but I am fairly confident that this is the most efficient way. 

We have discussed making the templates uniform, using either Jinja or  Django templates sitewide. I cannot remember which was the recommendation by @xtine at  18F. Once this is done, sharing content across the apps would be simpler.
